osu.Game.Rulesets.Osu/Mods/OsuModTransform.cs

```diff
@@ -30,38 +35,92 @@ internal class OsuModTransform : ModWithVisibilityAdjustment
+                        case DrawableSliderHead:
+                        case DrawableSliderTail:
+                        case DrawableSliderTick:
+                        case DrawableSliderRepeat:
+                            return;
 
-                default:
-                    var hitObject = (OsuHitObject)drawable.HitObject;
+                        default:
+                            var hitObject = (OsuHitObject)drawable.HitObject;
 
-                    float appearDistance = (float)(hitObject.TimePreempt - hitObject.TimeFadeIn) / 2;
+                            float appearDistance = (float)(hitObject.TimePreempt - hitObject.TimeFadeIn) / 2;
 
-                    Vector2 originalPosition = drawable.Position;
-                    Vector2 appearOffset = new Vector2(MathF.Cos(theta), MathF.Sin(theta)) * appearDistance;
+                            Vector2 originalPosition = drawable.Position;
+                            Vector2 appearOffset = new Vector2(MathF.Cos(theta), MathF.Sin(theta)) * appearDistance;
 
-                    // the - 1 and + 1 prevents the hit objects to appear in the wrong position.
-                    double appearTime = hitObject.StartTime - hitObject.TimePreempt - 1;
-                    double moveDuration = hitObject.TimePreempt + 1;
+                            // the - 1 and + 1 prevents the hit objects to appear in the wrong position.
+                            double appearTime = hitObject.StartTime - hitObject.TimePreempt - 1;
+                            double moveDuration = hitObject.TimePreempt + 1;
 
-                    using (drawable.BeginAbsoluteSequence(appearTime))
-                    {
-                        drawable
-                            .MoveToOffset(appearOffset)
-                            .MoveTo(originalPosition, moveDuration, Easing.InOutSine);
+                            using (drawable.BeginAbsoluteSequence(appearTime))
+                            {
+                                drawable
+                                    .MoveToOffset(appearOffset)
+                                    .MoveTo(originalPosition, moveDuration, Easing.InOutSine);
+                            }
+
+                            theta += (float)hitObject.TimeFadeIn / 1000;
+                            break;
                     }
 
-                    theta += (float)hitObject.TimeFadeIn / 1000;
                     break;
+
+                case MoveType.Emit:
+                    switch (drawable)
+                    {
+                        case DrawableSliderTick:
+                        case DrawableSliderRepeat:
+                            return;
+
+                        default:
+
+                            var hitObject = (OsuHitObject)drawable.HitObject;
+                            Vector2 origin = drawable.Position;
+
+                            if (hitObject is SliderRepeat || hitObject is SliderTailCircle)
+                                return;
```

Comment:
This is confusing. `DrawableSliderRepeat` is apparently already discarded above, so I'm not sure what the purpose of checking `SliderRepeat` here is. `DrawableSliderTick` is discarded above, but `SliderTick` is not checked here; conversely, `SliderTailCircle` is checked here but `DrawableSliderTail` is not discarded above. What is going on here?

Reply:
deleted